iceoryx_posh/doc/doxyfile-posh

```diff
@@ -1,2410 +0,0 @@
-# Doxyfile 1.8.6
```

Comment:
I'm using this file for the website (iox-#136-mkdocs-export-for-website) Doxygen will be displayed there, too. You deleted it to build Doxygen via CMake? Currently I think Doxybook [needs](https://github.com/matusnovak/doxybook2#hello-world) a Doxyfile.

Reply:
Good point,
what i understand is that doxybook needs the generated doxygen documentation as xml format:
From [https://github.com/matusnovak/doxybook2#command-line-arguments](https://github.com/matusnovak/doxybook2#command-line-arguments)
```
    -i, --input
        Path to the generated Doxygen XML folder. Must contain index.xml!
```
So the Doxyfile itself is not necessarily needed here. I removed the files because it's repetitive code which can be generated.
I added the flags to generate xml in the `IceoryxDoxygen.cmake` under iceoryx_utils/cmake.

Please note that the generated doxygen documentation is not one iceoryx documentation because it is split up into the different iceoryx component.